pkg/antctl/raw/multicluster/create/access_token.go

```diff
@@ -59,78 +70,174 @@ func (o *accessTokenOptions) validateAndComplete() error {
+		return createErr
+	} else {
+		fmt.Fprintf(cmd.OutOrStdout(), "RoleBinding \"%s\" created\n", roleBinding.Name)
+		createdRes = append(createdRes, u)
+	}
+
+	secretName := args[0]
+	secret := newSecret(secretName, memberTokenOpts.serviceAccount, memberTokenOpts.namespace)
+	createErr = k8sClient.Create(context.TODO(), secret)
+	if createErr != nil {
+		if !errors.IsAlreadyExists(createErr) {
+			fmt.Fprintf(cmd.OutOrStderr(), "Secret \"%s\" already exists\n", secretName)
+		}
+		fmt.Fprintf(cmd.OutOrStdout(), "Failed to create Secret \"%s\", start rollback\n", secretName)
+		return createErr
+	}
+	// It will take one or two seconds to wait for the Data.token to be created.
+	if err := waitForSecretReady(k8sClient, secretName, memberTokenOpts.namespace); err != nil {
 		return err
 	}
 
-	serviceAccount := corev1.ServiceAccount{
+	if err := k8sClient.Get(context.TODO(), types.NamespacedName{Name: secretName, Namespace: memberTokenOpts.namespace}, secret); err != nil {
+		return err
+	}
+	s := &corev1.Secret{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "Secret",
+		},
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      accessTokenOpts.serviceAccount,
-			Namespace: accessTokenOpts.namespace,
+			Name:      secretName,
+			Namespace: "member-cluster-namespace",
 		},
+		Data: secret.Data,
+		Type: corev1.SecretTypeOpaque,
 	}
 
-	var createErr error
-	fmt.Fprintf(cmd.OutOrStdout(), "Creating ServiceAccount \"%s\"\n", accessTokenOpts.serviceAccount)
-	createErr = k8sClient.Create(context.TODO(), &serviceAccount)
-	if createErr != nil {
-		if errors.IsAlreadyExists(createErr) {
-			fmt.Fprintf(cmd.OutOrStderr(), "ServiceAccount \"%s\" already exists\n", accessTokenOpts.serviceAccount)
-			createErr = nil
-		} else {
-			fmt.Fprintf(cmd.OutOrStdout(), "Failed to create ServiceAccount \"%s\", error: %s\n", accessTokenOpts.serviceAccount, createErr.Error())
-			return createErr
+	if memberTokenOpts.output != "" {
+		outputFile, err := os.OpenFile(memberTokenOpts.output, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0600)
+		if err != nil {
+			return err
 		}
-	} else {
-		fmt.Fprintf(cmd.OutOrStdout(), "ServiceAccount \"%s\" created\n", serviceAccount.Name)
-		defer func() {
-			if createErr != nil {
-				err := k8sClient.Delete(context.TODO(), &serviceAccount)
-				if err != nil {
-					fmt.Fprintf(cmd.OutOrStdout(), "Failed to delete ServiceAccount \"%s\", error: %s\n", serviceAccount.Name, err.Error())
-				} else {
-					fmt.Fprintf(cmd.OutOrStdout(), "ServiceAccount \"%s\" deleted\n", serviceAccount.Name)
+		defer outputFile.Close()
+
+		b, err := yaml.Marshal(s)
+		if err != nil {
+			return err
+		}
+		if _, err := outputFile.Write(b); err != nil {
+			return err
+		}
+		fmt.Fprintf(cmd.OutOrStdout(), "Member token saved to %s\n", memberTokenOpts.output)
+	}
+	fmt.Fprintf(cmd.OutOrStdout(), "You can now join any number of clusters by running the \"antctl mc join\" command on each member cluster with the token %s\n", s.Name)
```

Comment:
"You can now run the "antctl mc join" command with the token to have the cluster join the ClusterSet". I feel no need to print the token name.